### ISSUE TITLE
Allow include of other complete omni modules

### ISSUE THREAD

**Issue by vitreo12** [vitreo12/omni#75]
`Sine.oi:`

```
ins 1
outs 1

init:
    phase = 0.0

sample:
    increment = in1 / samplerate
    out1 = sin(phase * 2 * PI)
    phase = (phase + increment) % 1.0
```

`Something.oi:`

```
import "Sine.oi"

ins 1
outs 1

init:
    sine = Sine.init

sample:
    out1 = sine.perform(in1)
````

**Comment by vitreo12** [comment]
What about multichannel? Should it return a tuple(float, float...)?

```
sample:
    b = TwoChannels.perform
    out1 = b[0]
    out2 = b[1]
```

**Comment by vitreo12** [comment]
Not so sure about this: how to deal with the `perform` / `sample` difference?

**Comment by vitreo12** [comment]
Solution:
Still only have one final `.omni` object BUT, thanks to `require`, it can import other modules where specific definitions are provided. This could also allow to define a special syntax for chaining operators together, by simply checking if certain things (like, `init` and `perform` functions)  are defined for the specified `struct`. If `init` is not defined, use the standard `struct` constructor.

**Saw.omni:**
```nim
struct Saw:
    phase
    prev_value

def perform(saw Saw, freq_in = 440.0):
    freq = abs(freq_in)
    if freq == 0.0:
        freq = 0.01
    
    #0.0 would result in 0 / 0 -> NaN
    if saw.phase == 0.0:
        saw.phase = 1.0

    #BLIT
    n = trunc((samplerate * 0.5) / freq)
    phase_2pi = saw.phase * 2 * PI
    blit = 0.5 * (sin(phase_2pi * (n + 0.5)) / (sin(phase_2pi * 0.5)) - 1.0)

    #Leaky integrator
    freq_over_samplerate = (freq * 2 * PI) / samplerate * 0.25
    out_value = (freq_over_samplerate * (blit - saw.prev_value)) + saw.prev_value
    
    #Update entries in struct
    saw.phase += (freq / (samplerate - 1)) % 1.0
    saw.prev_value = out_value

    return out_value
```

**Osc.omni**
```nim
ins: 1
outs: 1

init:
    saw = Saw()

sample:
    out1 = saw.perform(in1)
```

**Comment by vitreo12** [comment]
Fixed with #84
